### PR TITLE
feat(suite): Add passphrase banner to dashboard

### LINKED ISSUE #12396: Add promo banner to dashboard (view-only)
**Ondra will design all banners first so we can unify it to common component.**

Screenshot 2024-05-15 at 10 30 25

https://www.figma.com/design/wGshbwWuR7shAA1i1UBqR8/branch/1O5PSi8KzMpeVyFMrEMZeA/Remember-Mode?node-id=2354-40396&t=03r75bJYrFjp9AqS-0

### PR DESCRIPTION
**Pull request by jvaclavik**
## Description

image

## Related Issue

Resolve #12396

### CODE REVIEW

File: packages/suite/src/views/dashboard/components/DashboardPassphraseBanner.tsx [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+    const device = useSelector(selectDevice);
+    const { isDiscoveryRunning } = useDiscovery();
+
+    if (
+        isDashboardPassphraseBannerVisible === false ||
+        device?.useEmptyPassphrase === true ||
+        isDiscoveryRunning === true ||
+        isDiscoveryRunning === undefined
+    )
+        return null;
```
Comment (peter-sanderson) [inline packages/suite/src/views/dashboard/components/DashboardPassphraseBanner.tsx:29]: it's multi-line `{}` :pray:

File: packages/suite/src/views/dashboard/components/DashboardPassphraseBanner.tsx [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+
+export const DashboardPassphraseBanner = () => {
+    const dispatch = useDispatch();
+    const { isDashboardPassphraseBannerVisible } = useSelector(selectSuiteFlags);
+    const device = useSelector(selectDevice);
+    const { isDiscoveryRunning } = useDiscovery();
+
+    if (
+        isDashboardPassphraseBannerVisible === false ||
+        device?.useEmptyPassphrase === true ||
```
Comment (peter-sanderson) [inline packages/suite/src/views/dashboard/components/DashboardPassphraseBanner.tsx:24]: I think the logic here shall be different :thinking: maybe?

This way the banner will be visible in EVERY passphrase wallet regardles if user uses the default one as well. Is this intended?

I think, condition shall be: if ALL your devices are `useEmptyPassphrase===false`  the we shall show banner? Not sure if this is enough, maybe lets discuss this.

File: packages/suite/src/views/settings/SettingsDebug/ViewOnlySettings.tsx [hunk 3]
```diff
@@ -47,6 +51,23 @@ export const ViewOnlySettings = () => {
                     />
                 </ActionColumn>
             </SectionItem>
+
+            <SectionItem>
+                <TextColumn title="Set isDashboardPassphraseBannerVisible" />
+                <ActionColumn>
+                    <Checkbox
+                        isChecked={isDashboardPassphraseBannerVisible}
+                        onClick={() => {
```
Comment (peter-sanderson) [inline packages/suite/src/views/settings/SettingsDebug/ViewOnlySettings.tsx:60]: nit: separate to handleOnClicnt or something

**Review by peter-sanderson — CHANGES_REQUESTED** [review]
Lets talk ;)

File: packages/suite/src/views/dashboard/components/DashboardPassphraseBanner.tsx (hunk 2 above)
Comment (jvaclavik) [reply packages/suite/src/views/dashboard/components/DashboardPassphraseBanner.tsx:24]: I don't think that it should be only when all have `useEmptyPassphrase===false`

But we need @Hermez-cz for decide this

Anyway, we can merge it this way and update it later. I also added another condition so the banner will not be visible anymore when your default wallet is already passphrase.

**Review by peter-sanderson — APPROVED** [review]

**Pull request merged**